#164608984 add feature to get edit history for a comment

danprocoder opened the pull request:
#### What does this PR do?
Add feature to get edit history for a comment

#### Description of Task to be completed?
- Created feature to get edit history of a comment.
  GET /api/articles/<slug>/comment/<comment-id>/history
  * Authorization required
  * Admin priviledge required
- Wrote tests

#### How should this be manually tested?
1. Clone this branch.
2. Run `npm install` to install the dependencies.
3. Run `npm test` to run the tests.

#### What are the relevant pivotal tracker stories?
[#164608984](https://pivotaltracker.com/story/show/164608984)

danprocoder committed:
feat(editCommentHistory): add feature to get edit history for a comment
- Created feature to get edit history of a comment.
  GET /api/articles/<slug>/comment/<comment-id>/history
  * Authorization required
  * Admin priviledge required
- Wrote tests
[Finishes #164608984]

danprocoder committed:
feat(editCommentHistory): add feature to get edit history for a comment
- Created feature to get edit history of a comment.
  GET /api/articles/<slug>/comment/<comment-id>/history
  * Authorization required
  * Admin priviledge required
- Wrote tests
[Finishes #164608984]

danprocoder committed:
Merge branch 'ft-view-comment-edit-history-164608984' of https://github.com/andela/artemis-ah-backend into ft-view-comment-edit-history-164608984

Dsalz reviewed (commented):
Your code commenting game is fire but the user who created the comment should also have access to the edit history not just the admin. 

So instead of just granting admin access you can check if the person who made the request is the owner of the comment and grant them access as well

danprocoder committed:
Increased test coverage

danprocoder committed:
feat(commentEditHistory): change access priviledge

Only an admin AND the user that created the comment can see the edit history of that comment

[Finishes #164608984]

danprocoder committed:
feat(commentEditHistory): change access priviledge

Only an admin AND the user that created the comment can see the edit history of that comment

[Finishes #164608984]

danprocoder committed:
Merge branch 'ft-view-comment-edit-history-164608984' of https://github.com/andela/artemis-ah-backend into ft-view-comment-edit-history-164608984
